.github/workflows/unit-test-on-pull-request.yml

```
@@ -23,83 +21,99 @@ jobs:
       - name: Linter
         run: |
           go version
-          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.56.2
-          make lint
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh \
+            | sh -s -- -b $(go env GOPATH)/bin v1.56.2
+          make lint TARGET_ARCH=${{ matrix.target_arch }}
 
   test:
-    name: Test
+    name: Test (${{ matrix.target_arch }})
     runs-on: ubuntu-24.04
     strategy:
-      fail-fast: true
-      max-parallel: 2
       matrix:
-        go: ["stable"]
+        target_arch: [amd64, arm64]
     steps:
       - name: Clone code
         uses: actions/checkout@v4
       - name: Set up environment
         uses: ./.github/workflows/env
-      - name: Build
-        run: |
-          echo $PATH
-          make test
       - name: Tests
-        run: |
-          make test
+        run: make test TARGET_ARCH=${{ matrix.target_arch }}
 
   build-integration-test-binaries:
-    name: Build integration test binaries
+    name: Build integration test binaries (${{ matrix.target_arch }})
     runs-on: ubuntu-24.04
     timeout-minutes: 10
+    strategy:
+      matrix:
+        target_arch: [amd64, arm64]
     steps:
       - name: Clone code
         uses: actions/checkout@v4
       - name: Set up environment
         uses: ./.github/workflows/env
       - name: Prepare integration test binaries for qemu tests
-        run: |
-          make integration-test-binaries
+        run: make integration-test-binaries TARGET_ARCH=${{ matrix.target_arch }}
       - name: Upload integration test binaries
         uses: actions/upload-artifact@v4
         with:
-          name: integration-test-binaries
+          name: integration-test-binaries-${{ matrix.target_arch }}
           path: support/*.test
 
   integration-tests:
-    name: Integration tests (kernel v${{ matrix.kernel-version }})
+    name: Integration tests (kernel v${{ matrix.kernel }} ${{ matrix.target_arch }})
```

Nit: the Github UI currently makes it hard to read this name and so it shows at the moment something like `Integration tests (kernel v4.19.3...` in the [side panel](https://github.com/elastic/otel-profiling-agent/pull/94/checks).

Shorten the shown text to easier spot the differences between the test runs:
```suggestion
    name: Integration test (v${{ matrix.kernel }} ${{ matrix.target_arch }})
```